src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AnnotationUtils.java

```diff
@@ -36,7 +42,51 @@ public static String encodeValueList(final List<Double> valueList, final String
      * @return a comma-separated String
      */
     public static String encodeStringList( final List<String> stringList) {
-        return StringUtils.join(stringList, ",");
+        return StringUtils.join(stringList, LIST_DELIMITER);
+    }
+
+    /**
+     * Helper function to convert a List of Strings to a comma-separated String
+     * @param somethingList the ArrayList with String data
+     * @return a comma-separated String
+     */
+    public static String encodeAnyASList( final List<?> somethingList) {
+        return StringUtils.join(somethingList, ALLELE_SPECIFIC_PRINT_DELIM).replaceAll(BRACKET_REGEX, "");  //Who actually wants brackets at the ends of their string?  Who???
     }
 
+    /**
+     * Helper function to determine if an annotation is allele-specific
+     * @param annotation the annotation to be tested
+     * @return true if the annotation is expected to have values per-allele
+     */
+    public static boolean isAlleleSpecific(final InfoFieldAnnotation annotation) {
+        if (annotation instanceof AS_RankSumTest) {
+            return true;
+        }
+        if (annotation instanceof AS_StrandBiasTest) {
+            return true;
+        }
+        if (annotation instanceof AS_RMSMappingQuality) {
+            return true;
+        }
+        if (annotation instanceof AS_StandardAnnotation) {
+            return true;
```

Comment:
Are these really all the allele-specific annotations? I thought that there were more than 4...

Maybe you should just check if the class name starts with "AS_" instead?

Reply:
Well, there are two AS_StrandBiasTests and two AS_RankSumTests.

I'd strongly prefer to keep it this way.  The other way just seems too sketchy.  It shouldn't be too costly because it shouldn't be called a million times